File: cpp/Server-UUID.cpp
#include <Ice/Ice.h>
#include "PrinterI.h"

using namespace std;
using namespace Ice;

class Server: public Application {
  int run(int argc, char* argv[]) {
    Example::PrinterPtr servant = new Example::PrinterI();

    ObjectAdapterPtr adapter =
         communicator()->createObjectAdapter("PrinterAdapter");
    ObjectPrx proxy = adapter->addWithUUID(servant);

    cout << communicator()->proxyToString(proxy) << endl;

    adapter->activate();
    shutdownOnInterrupt();
    communicator()->waitForShutdown();

    return 0;
  }
};

int main(int argc, char* argv[]) {
  Server app;
  return app.main(argc, argv);
}

File: py/server-uuid.py
#!/usr/bin/env -S python3 -u

import sys
import Ice
Ice.loadSlice('printer.ice')
import Example


class PrinterI(Example.Printer):
    n = 0

    def write(self, message, current=None):
        print("{0}: {1}".format(self.n, message))
        sys.stdout.flush()
        self.n += 1


def main(ic):
    servant = PrinterI()
    adapter = ic.createObjectAdapter("PrinterAdapter")
    proxy = adapter.addWithUUID(servant)

    print(proxy, flush=True)

    adapter.activate()
    ic.waitForShutdown()


if __name__ == "__main__":
    try:
        with Ice.initialize(sys.argv) as ic:
            main(ic)
    except KeyboardInterrupt:
        pass
